routes/teams.js

```diff
@@ -1,105 +1,126 @@
+        callback(items, res);
       }
-    })
+    }
+    else {  // data not already in Mongo
+      var players = fetchPlayersFromApi(team_id,res,callback)
+    }
   });
-
 }
 
 
-var formatTeams = function(hierarchy_response){
-  for (i=0;i<hierarchy_response.conferences.length;i++){
-    for (j=0;j<hierarchy_response.conferences[i].divisions.length;j++){
-      for(k=0; k< hierarchy_response.conferences[i].divisions[j].teams.length; k++){
-            teams.push(hierarchy_response.conferences[i].divisions[j].teams[k]);
+
+var fetchPlayersFromApi = function(team_id,res,callback){
+var json_response = '';
+var players = {};
+  request('https://api.sportsdatallc.org/nba-t3/seasontd/2014/reg/teams/'+team_id+'/statistics.json?api_key='+config.nba_key, function (error, response, body) {
+    if (!error && response.statusCode == 200) {
+      json_response = JSON.parse(body);
+      players = formatPlayers(json_response, team_id);
+      mongoInsertPlayers(team_id, players);
+      callback(players, res)
+    } 
+  }); 
+}
+
+// flatten players, average, & total objects into one level
+var formatPlayers = function(response, team_id){
+  playersarray = [];
+  for (i=0;i<response.players.length;i++){
+    playersarray[i] = {};
+    for(var key in response.players[i]){
+      if (key != "total" && key != "average"){
+        var value = response.players[i][key];
+        playersarray[i][key] = value;
+      }
+      else if (key =="total"){
+        for(var key in response.players[i].total) {
+          var value = response.players[i].total[key];
+          playersarray[i]["total_"+key] = value;
+        }
       }
-    }   
+      else {
+        for(var key in response.players[i].average) {
+          var value = response.players[i].average[key];
+          playersarray[i]["average_"+key] = value;
+        }
+      }
+    }
   }
-return teams
+  var team = formatPlayersDocument(team_id, playersarray);
+  return team;
 }
-//this is for the /teams page search field ajax
-router.get('/team', function(req, res) {
-    var term = req.query.q;
-    console.log(term);
+  
 
-    //find the team, This syntax does a sql-type like clause with case insensitivy(sp???) with the RegEx
-    db.collection('teams').find({$or: [{'name': new RegExp(term, 'i')}, {'market': new RegExp(term, 'i')}]}).sort({'market': 1}).toArray(function (err, items) {
-    	res.json(items);
+var formatPlayersDocument = function(team_id, players){
+  teamDocument = {};
+  teamDocument["team_id"] = team_id;
+  teamDocument["players"] = players; 
+  return teamDocument;
+}
+
+
+function mongoInsertPlayers(team_id, team_document){
+  console.log("inserting into the DB");
+  db.open(function(err, db){
+    db.collection("players").update({team_id: team_id},
+    {$set: {team_id: team_document["team_id"], last_updated: new Date(), players: team_document["players"]}},
+    {upsert: true, multi:false}, function (err, upserted){
+      console.log("upsert complete");
```

Comment:
id be more interested in outputting when an error occurs, i expect/hope that this would succeed.